cpp/modmesh/buffer/pymod/wrap_SimpleArrayPlex.cpp

```diff
@@ -75,133 +202,47 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArrayPlex : public WrapBase<Wr
-            if (!pybind11::isinstance<pybind11::int_>(value))
-            {
-                throw std::runtime_error("Data type mismatch, expected Python int");
-            }
-            auto * array = static_cast<SimpleArrayInt64 *>(array_plex.mutable_instance_ptr());
-            array->fill(value.cast<int64_t>());
-            break;
-        }
-        case DataType::Uint8:
-        {
-            if (!pybind11::isinstance<pybind11::int_>(value))
-            {
-                throw std::runtime_error("Data type mismatch, expected Python int");
-            }
-            auto * array = static_cast<SimpleArrayUint8 *>(array_plex.mutable_instance_ptr());
-            array->fill(value.cast<uint8_t>());
-            break;
-        }
-        case DataType::Uint16:
-        {
-            if (!pybind11::isinstance<pybind11::int_>(value))
-            {
-                throw std::runtime_error("Data type mismatch, expected Python int");
-            }
-            auto * array = static_cast<SimpleArrayUint16 *>(array_plex.mutable_instance_ptr());
-            array->fill(value.cast<uint16_t>());
-            break;
-        }
-        case DataType::Uint32:
-        {
-            if (!pybind11::isinstance<pybind11::int_>(value))
-            {
-                throw std::runtime_error("Data type mismatch, expected Python int");
-            }
-            auto * array = static_cast<SimpleArrayUint32 *>(array_plex.mutable_instance_ptr());
-            array->fill(value.cast<int32_t>());
-            break;
-        }
-        case DataType::Uint64:
-        {
-            if (!pybind11::isinstance<pybind11::int_>(value))
-            {
-                throw std::runtime_error("Data type mismatch, expected Python int");
-            }
-            auto * array = static_cast<SimpleArrayUint64 *>(array_plex.mutable_instance_ptr());
-            array->fill(value.cast<int64_t>());
-            break;
-        }
-        case DataType::Float32:
-        {
-            if (!pybind11::isinstance<pybind11::float_>(value))
-            {
-                throw std::runtime_error("Data type mismatch, expected Python float");
-            }
-            auto * array = static_cast<SimpleArrayFloat32 *>(array_plex.mutable_instance_ptr());
-            array->fill(value.cast<float>());
-            break;
-        }
-        case DataType::Float64:
-        {
-            if (!pybind11::isinstance<pybind11::float_>(value))
-            {
-                throw std::runtime_error("Data type mismatch, expected Python float");
-            }
-            auto * array = static_cast<SimpleArrayFloat64 *>(array_plex.mutable_instance_ptr());
-            array->fill(value.cast<double>());
-            break;
-        }
-        default:
-            throw std::runtime_error("Unsupported datatype");
-        }
+        wrapped_type array_plex(shape, datatype_str);
+        auto datatype = array_plex.data_type();
+        execute_callback_with_typed_array(
```

Comment:
`execute_callback_with_typed_array` is used.

Reply:
Beautiful.